include/vast/Translation/CodeGenContext.hpp

```diff
@@ -161,6 +161,9 @@ namespace vast::cg
                 if (llvm::isa< clang::FunctionDecl >(dctx))
                     continue;
 
+                if (llvm::isa< clang::LinkageSpecDecl >(dctx))
```

Comment:
How is this related to the unsupported dialect? I mean having some extra small-ish change(s) thrown in a bigger PR is fine, but the commit should be self-contained and ideally mention it in the PR description.

Reply:
If you are walking the decl context parent, you may find `LinkageSpecDecl` as one of the parent and not a `NamedDecl`. That will lead to reaching the unreachable. 

I will move it in the separate commit under this PR to avoid mixup with other changes.